janusgraph-core/src/main/java/org/janusgraph/diskstorage/indexing/IndexMutation.java

```diff
@@ -68,22 +78,24 @@ public void resetDelete() {
         isDeleted=false;
     }
 
-    public static final Function<IndexEntry,String> ENTRY2FIELD_FCT = new Function<IndexEntry, String>() {
-        @Nullable
-        @Override
-        public String apply(final IndexEntry indexEntry) {
-            return indexEntry.field;
-        }
-    };
+    private boolean isCollection(String field) {
+        KeyInformation keyInformation = storeRetriever.get(field);
+        return keyInformation != null && keyInformation.getCardinality() != Cardinality.SINGLE;
+    }
+
+    private final Function<IndexEntry, AbstractMap.SimpleEntry<String, Object>> entryConversionFunction =
+            indexEntry -> isCollection(Objects.requireNonNull(indexEntry).field) ?
+                    new AbstractMap.SimpleEntry<>(indexEntry.field, indexEntry.value) :
+                    new AbstractMap.SimpleEntry<>(indexEntry.field, DUMMY);
```

Comment:
I think we don't need to wrap `indexEntry.field` into `AbstractMap.SimpleEntry`. I.e. we can run like this:
```
private final Function<IndexEntry, Object> entryConversionFunction =
    indexEntry -> isCollection(Objects.requireNonNull(indexEntry).field) ?
        new AbstractMap.SimpleEntry<>(indexEntry.field, indexEntry.value) :
        indexEntry.field;
```